### ISSUE TITLE
Failed executions are not getting saved in queue mode

### ISSUE THREAD

**Issue by phizab** [n8n-io/n8n#7705]
**Describe the bug**
Saving failed executions does not work when using queue mode and if saving of successful executions is disabled. This does not apply to sub workflows. These are getting saved.

**To Reproduce**

1. Create a workflow with "Schedule Trigger" (e.g. 10 seconds interval) and "Stop and Error" node (any message).
2. Set "Save failed production executions" to "Save" and "Save successful production executions" to "Do not save".
3. Activate the workflow.

**Expected behavior**
Using the workflow above the executions, which always fail because of the "Stop and Error" node, should be listed in executions list. Instead they getting deleted. But when changing "Save successful production executions" to "Save", they appear in executions list.

**Environment (please complete the following information):**

- Docker-Image with n8n Version 1.15.2 (tested with 1.16.0, too)
- Database system: MariaDB (yep, I know...) / SQLite (in test workflow)
- Operation mode: queue

**Additional context**
I used following docker compose to reproduce the problem:

```yaml
services:
  n8n:
    image: n8nio/n8n:1.15.2
    environment:
      - GENERIC_TIMEZONE=Europe/Berlin
      - TZ=Europe/Berlin
      - EXECUTIONS_MODE=queue
      - QUEUE_BULL_REDIS_HOST=redis
    volumes:
      - .n8n/:/home/node/.n8n/
    ports:
      - 5678:5678
  
  n8n-worker:
    image: n8nio/n8n:1.15.2
    command: [ "worker", "--concurrency", "30" ]
    environment:
      - GENERIC_TIMEZONE=Europe/Berlin
      - TZ=Europe/Berlin
      - EXECUTIONS_MODE=queue
      - QUEUE_BULL_REDIS_HOST=redis
    volumes:
      - ./.n8n:/home/node/.n8n
    depends_on:
      - redis
  
  redis:
    image: redis:6.2-alpine
    command: [ "redis-server", "--appendonly", "yes" ]
    volumes:
      - redis-data:/data

volumes:
  redis-data:
```

Workflow for test:

```json
{
  "meta": {
    "instanceId": "93dc80749bba53c076d9f2b347ca4cfec2db7f6d91ab697f3860c9cf17844726"
  },
  "nodes": [
    {
      "parameters": {
        "rule": {
          "interval": [
            {
              "field": "seconds",
              "secondsInterval": 10
            }
          ]
        }
      },
      "id": "4e5db2e4-e173-412a-ba1c-1eafe4a81c04",
      "name": "Schedule Trigger",
      "type": "n8n-nodes-base.scheduleTrigger",
      "typeVersion": 1.1,
      "position": [
        760,
        340
      ]
    },
    {
      "parameters": {
        "errorMessage": "Test!"
      },
      "id": "0caf70ef-37c2-4dcf-aadb-ff1ddd3c4b97",
      "name": "Stop and Error",
      "type": "n8n-nodes-base.stopAndError",
      "typeVersion": 1,
      "position": [
        980,
        340
      ]
    }
  ],
  "connections": {
    "Schedule Trigger": {
      "main": [
        [
          {
            "node": "Stop and Error",
            "type": "main",
            "index": 0
          }
        ]
      ]
    }
  }
}
```

**Comment by ivov** [comment]
Thank you for the detailed report. This should be fixed by https://github.com/n8n-io/n8n/pull/7634 released with 1.16. Can you please upgrade and see if the issue persists?

**Comment by phizab** [comment]
Even with 1.16.0 I have this problem. Just used docker image n8nio/n8n:1.16.0 in docker compose from above.

**Comment by ivov** [comment]
Tracking here: https://linear.app/n8n/issue/PAY-1010

**Comment by janober** [comment]
Fix got released with [`n8n@1.18.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.18.0)

**Comment by Joffcom** [comment]
Hey @phizab,

It looks like this has been fixed now, Let me know how you get on after an update and if it is still causing issues let me know and we can open this again.
